server/routes/manageConditions/types/additionalConditionInputs/curfewTerms.ts

```diff
@@ -27,4 +79,28 @@ class CurfewTerms {
   alternativeReviewPeriod: string
 }
 
+const splitTimes = (value: SimpleTime) => {
+  if (!Array.isArray(value.hour)) {
+    return [value]
+  }
+  return value.hour.map((h: string, index) => {
+    const hour = h.length === 1 ? `0${h}` : h
+    const minute = value.minute[index].length === 1 ? `0${value.minute[index]}` : value.minute[index]
+    return SimpleTime.fromString(`${hour}:${minute} ${value.ampm[index]}`)
+  })
+}
+
+const selectRelevantEntry = (numberOfCurfews: string, enteredValues: SimpleTime[]) => {
+  if (numberOfCurfews === 'One curfew') {
+    return enteredValues[enteredValues.length - 3]
+  }
+  if (numberOfCurfews === 'Two curfews') {
+    return enteredValues[enteredValues.length - 2]
+  }
+  if (numberOfCurfews === 'Three curfews') {
+    return enteredValues[enteredValues.length - 1]
+  }
+  return undefined
+}
```

Comment:
With the `SimpleTime` objects built (via `splitTimes` above), all but one of the created objects will be invalid, as it will have been built with `undefined` entries. As the arrays of `SimpleTime`s are in order though, we can always tell which one we should use based on the selected number of curfews.
For example, when `Two curfews` is selected (and entered), `enteredValues` will look like this:
![image](https://github.com/ministryofjustice/create-and-vary-a-licence/assets/102144566/0afb9e6d-aeab-4bbc-90e4-c0536f76ae8d)
So we can use `enteredValues.length - 2` to consistently select the `ValidSimpleTime`. The same logic applies to the other cases.